spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java

```diff
@@ -169,7 +170,7 @@ public Dataset<Row> expire() {
       Dataset<Row> validFiles = buildValidFileDF(ops.refresh());
 
       // determine expired files
-      this.expiredFiles = originalFiles.except(validFiles);
+      this.expiredFiles = originalFiles.except(validFiles).unpersist();
```

Comment:
Again I believe if you want to get rid of the cache I do think you need to call that the allManifests table